### PR TITLE
docs: Hardening BYOB

### PR DESCRIPTION
**Pull request by laurentsimon**

### CODE REVIEW

**Comment by laurentsimon** [comment]
Ready for review. Once this is merged I'll verify that the instruction and code examples work as expected - including the use of the new Action defined in this PR

File: BYOB.md [hunk 1]
```diff
@@ -219,16 +223,55 @@ The last thing to do in the TCA is to [generate the metadata layout file](https:
...
+- [Download the artifacts](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L142-L149) uploaded by the TCA.
+- [Publish the artifacts](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L151-L168).
+
+### Best SDLC Practices
+
+It is important you follow best development practices for your code, including your TRW, TCA and existing Action. In particular:
 
-TODO(Scorecard, pin actions and dependencies, verify provenance when downloading binary, etc, Use low-perm)
+- Harden your CI, e.g., declare your workflow permissions to `read-only`.
+- Pin your depenencies by hash, to avoid dependency confusion attacks and speed up incidence response.
```
Comment (ianlewis) [inline BYOB.md:275]: Except for the delegator workflows

File: BYOB.md [hunk 2]
```diff
@@ -219,16 +223,55 @@ The last thing to do in the TCA is to [generate the metadata layout file](https:
...
+- If you download binaries, verify their SLSA provenance before running them.
```
Comment (ianlewis) [inline BYOB.md:276]:
```suggestion
- If you download binaries, verify their SLSA provenance before running them. Use the [`installer`](https://github.com/slsa-framework/slsa-verifier/tree/main/actions/installer) action to install and use `slsa-verifier`.
```

**Review by ianlewis — APPROVED** [review]

File: BYOB.md [hunk 3]
```diff
@@ -88,7 +92,7 @@ The SRW acts as the build's orchestrator. It calls the TCA, generates provenance
...
+In this example, we will assume there is an existing [GitHub Action](https://github.com/laurentsimon/byob-doc/blob/main/action.yml) which builds an artifact. The Action is fairly simple: it just [echos the parameters into the artifact](https://github.com/laurentsimon/byob-doc/blob/main/action.yml#L58). It also takes a [username, password and token](https://github.com/laurentsimon/byob-doc/blob/main/action.yml#L31-L34) to retrieve / push information from a remote registry. Like popular release Actions, it [releases the built artifact to GitHub releases](https://github.com/laurentsimon/byob-doc/blob/main/action.yml#L67-L78). It outputs the [name of the built artifact and the status of the build](https://github.com/laurentsimon/byob-doc/blob/main/action.yml#L35-L41). See the full [action.yml](https://github.com/laurentsimon/byob-doc/blob/main/action.yml).
```
[the remark this answers is not in the document]
Comment (laurentsimon) [reply BYOB.md:95]: I'm planning to create tags and send a follow-up PR to use them

**Pull request merged**